File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int freq=0;
        int curr=0;
        for(int i=0;i<nums.size();i++)
        {
            if(curr==nums[i])
            {
                freq++;
            }
            else if(freq==0)
            {
                freq=1;
                curr=nums[i];
            }
            else
            {
                freq--;
            }
        }
        return curr;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count=0, ele=0;
        for(int i=0;i<nums.length;i++)
        {
            if(ele==nums[i])
            {
                count++;
            }
            else if(count==0)
            {
                ele = nums[i];
                count=1;
            }
            else
            {
                count--;
            }
        }
        return ele;
    }
}